fractions/microprofile/microprofile-opentracing/src/main/java/org/wildfly/swarm/mpopentracing/deployment/MicroprofileServerTracingFeature.java

```diff
@@ -0,0 +1,32 @@
+package org.wildfly.swarm.mpopentracing.deployment;
+
+import io.opentracing.contrib.jaxrs2.server.OperationNameProvider.ClassNameOperationName;
+import io.opentracing.contrib.jaxrs2.server.ServerTracingDynamicFeature;
+import io.opentracing.contrib.jaxrs2.server.ServerTracingDynamicFeature.Builder;
+import io.opentracing.util.GlobalTracer;
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * @author Pavol Loffay
+ */
+@Provider
+public class MicroprofileServerTracingFeature implements DynamicFeature {
+
+  private ServerTracingDynamicFeature delegate;
+
+  public MicroprofileServerTracingFeature() {
+    // TODO tracer CDI injection didn't work
+    this.delegate = new Builder(GlobalTracer.get())
```

Comment:
@mkouba do you think we can inject the tracer via CDI?

I have tried but it didn't work. Tracer was always null. I have tried constructor and field injection.

Reply:
I have no idea how `DynamicFeature` works and whether JAX-RS supports injection there. If there is a bean for  `GlobalTracer` then `CDI.current().select(GlobalTracer.class).get()` should work but using `CDI.current()` might be tricky in Java EE.

Reply:
There should be `Tracer.class` bean, I will try that, although is it safe to use?